src/_pytest/python.py

```diff
@@ -380,6 +381,11 @@ class _EmptyClass: pass  # noqa: E701
 # fmt: on
 
 
+def check_if_test_is_dynamically_parametrized(metafunc):
```

Comment:
i believe this has to be the last hook added, and if its always added, it ought to be part of a default implementation hookwrapper thats tryfirst

perhaps this is better as property of metafunc to begin with